src/SourceBuild/content/eng/tools/tasks/Microsoft.DotNet.SourceBuild.Tasks.SdkArchiveDiff/GetClosestOfficialSdk.cs

```diff
@@ -0,0 +1,73 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using System.IO;
+using System.Net;
+using System.Net.Http;
+using System.Threading;
+using System.Threading.Tasks;
+using Microsoft.Build.Framework;
+
+public class GetClosestOfficialSdk : Microsoft.Build.Utilities.Task, ICancelableTask
```

Comment:
There's a mix of Sdk specific functionality and non-SDK specific functionality here (or at least terminology). For instance, the comparison (find archive diffs) I think works for any archive. On the other hand, Archive has functionality that only works for the SDK (retrieving version numbers).

The below code mostly works for anything, except for `GetLatest`. Maybe it would be better to factor the common code below out into a base task class?

Reply:
Good idea, I extracted the non-specific functionality to `GetClosestArchive`